### PR TITLE
Frontend alignment with interest holder updates

### PR DESCRIPTION
**Pull request by FuriousLlama**
Updated front end models and fixed some issues related to interest holders

### CODE REVIEW

File: source/frontend/src/features/mapSideBar/acquisition/common/GenerateForm/hooks/useGenerateAgreement.ts [hunk 1]
```diff
@@ -31,9 +33,11 @@ export const useGenerateAgreement = () => {
...
       team => team.personProfileTypeCode === 'MOTILAWYER',
     );
-    const ownerSolicitor = file.acquisitionFileOwnerSolicitors?.length
-      ? file.acquisitionFileOwnerSolicitors[0]
-      : undefined;
+    const solicitors = file.acquisitionFileInterestHolders
+      ?.filter(x => x.interestHolderType?.id === InterestHolderType.OWNER_SOLICITOR)
+      .filter((x): x is Api_InterestHolder => !!x);
+    const ownerSolicitor =
+      solicitors !== undefined && solicitors.length > 0 ? solicitors[0] : undefined;
```
Comment (asanchezr) [inline source/frontend/src/features/mapSideBar/acquisition/common/GenerateForm/hooks/useGenerateAgreement.ts:40]: in **useGenerateExpropriationForm1** the following pattern was used. Would lean towards using the same here to keep consistency:

```tsx
    const ownerSolicitor = file.acquisitionFileInterestHolders?.find(
      x => x.interestHolderType?.id === InterestHolderType.OWNER_SOLICITOR,
    );
```

File: source/frontend/src/features/mapSideBar/acquisition/tabs/stakeholders/update/models.test.ts [hunk 2]
```diff
@@ -1,21 +1,44 @@
...
-  StakeHolderForm,
-} from './models';
+import { InterestHolderForm, StakeHolderForm } from './models';
+
+const emptyInterestHolderForm: InterestHolderForm = {
+  interestHolderId: null,
+  //person: null,
+  //personId: '',
+  //organization: null,
+  //organizationId: '',
```
Comment (asanchezr) [inline source/frontend/src/features/mapSideBar/acquisition/tabs/stakeholders/update/models.test.ts:12]: please remove dead code (or take out the comments)

File: source/frontend/src/features/mapSideBar/acquisition/tabs/stakeholders/update/models.ts [hunk 3]
```diff
@@ -13,112 +15,127 @@ export class StakeHolderForm {
...
+        return apiInterestHolder;
       })
-      .value();
+      .value()
+      .filter((x): x is Api_InterestHolder => x !== null);
+  }
+}
+
+export class InterestHolderPropertyFormModel {
+  //acquisitionFileProperty = {};
```
Comment (asanchezr) [inline source/frontend/src/features/mapSideBar/acquisition/tabs/stakeholders/update/models.ts:85]: please remove comment or re-enable the property

**Review by asanchezr — APPROVED** [review]

**Comment by devinleighsmith** [comment]
@FuriousLlama there is a 1 showing up here for some reason: 
image

File: source/frontend/src/features/mapSideBar/acquisition/common/GenerateForm/hooks/useGenerateAgreement.ts (hunk 1 above)
Comment (FuriousLlama) [reply source/frontend/src/features/mapSideBar/acquisition/common/GenerateForm/hooks/useGenerateAgreement.ts:40]: is doing the same, the difference is that the second one has potentially many interest holders

**Comment by FuriousLlama** [comment]
> //acquisitionFileProperty = {};

Checked and my local does not have that on the stakeholder screen. That might be something on your side

File: source/frontend/src/features/mapSideBar/acquisition/common/GenerateForm/hooks/useGenerateAgreement.ts (hunk 1 above)
Comment (asanchezr) [reply source/frontend/src/features/mapSideBar/acquisition/common/GenerateForm/hooks/useGenerateAgreement.ts:40]: The 2nd one still only pick the first one it finds- ie "solicitors[0]" so to me that is doing the same as "find" which returns the first it finds. But not a big deal. We can align that later

**Pull request merged**